src/platform_impl/macos/app_state.rs

```diff
@@ -368,13 +584,29 @@ impl AppState {
         HANDLER.events().push_back(wrapper);
     }
 
+    pub fn stop() {
+        let app = NSApp();
```

Comment:
Shouldn't it be inside the relaese pool? I'd assume it ref counts here?

Reply:
I just verbatim moved some existing code for this - I'm honestly not sure.

Reply:
previously this code was like:

```rust
         if HANDLER.should_exit() {
            let app = NSApp();
            autoreleasepool(|_| {
                app.stop(None);
                // To stop event loop immediately, we need to post some event here.
                app.postEvent_atStart(&NSEvent::dummy(), true);
            });
        }
```

The `NSApp` is weird global state so maybe there's something subtle related to that?

Reply:
It looks like `NSapp` returns an `Id` which implements `Drop` which calls `ffi::objc_release(self.ptr.as_ptr().cast())`, so it looks like it doesn't need to be collected by an auto release pool.